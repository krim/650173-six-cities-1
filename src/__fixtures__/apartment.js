export default {
  city: {
    name: `Amsterdam`,
    location: {
      latitude: 51.225402,
      longitude: 6.776314,
      zoom: 13
    }
  },
  previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
  images: [
    `https://es31-server.appspot.com/six-cities/static/hotel/18.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/20.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/15.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/8.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/5.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/7.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/19.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/14.jpg`
  ],
  title: `Waterfront with extraordinary view`,
  isFavorite: false,
  isPremium: true,
  rating: 2.7,
  type: `hotel`,
  bedrooms: 2,
  maxAdults: 8,
  price: 393,
  goods: [
    `Breakfast`,
    `Washer`,
    `Laptop friendly workspace`
  ],
  host: {
    id: 25,
    name: `Angelina`,
    isPro: true,
    avatarUrl: `img/avatar-angelina.jpg`
  },
  description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
  location: {
    latitude: 51.214402,
    longitude: 6.764314000000001,
    zoom: 16
  },
  id: 1
};
