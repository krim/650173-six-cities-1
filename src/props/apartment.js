import PropTypes from 'prop-types';
import {range} from 'react-range-proptypes';

export default PropTypes.shape({
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  previewImage: PropTypes.string.isRequired,
  rating: range(0, 100).isRequired,
  price: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
  bedrooms: PropTypes.number.isRequired,
  maxAdults: PropTypes.number.isRequired,
  isFavorite: PropTypes.bool.isRequired,
  isPremium: PropTypes.bool.isRequired,
  location: PropTypes.shape({
    latitude: PropTypes.number.isRequired,
    longitude: PropTypes.number.isRequired,
    zoom: PropTypes.number.isRequired
  }),
  host: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    isPro: PropTypes.bool.isRequired,
    avatarUrl: PropTypes.string.isRequired
  })
});
