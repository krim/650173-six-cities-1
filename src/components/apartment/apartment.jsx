import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';

import {apartmentProps} from '../../props';
import getRating from '../../libs/get-rating';

class Apartment extends PureComponent {
  render() {
    const {
      apartment,
      onClick,
      onMouseOver,
      onMouseOut
    } = this.props;

    return (
      <article
        onMouseOver={onMouseOver}
        onMouseOut={onMouseOut}
        className="cities__place-card place-card"
      >
        { apartment.isPremium && <div className="place-card__mark"><span>Premium</span></div> }
        <div className="cities__image-wrapper place-card__image-wrapper">
          <a href="#">
            <img
              className="place-card__image" src={apartment.previewImage} width="260" height="200" alt="Place image"
            />
          </a>
        </div>
        <div className="place-card__info">
          <div className="place-card__price-wrapper">
            <div className="place-card__price">
              <b className="place-card__price-value">&euro;{apartment.price}</b>
              <span className="place-card__price-text">&#47;&nbsp;night</span>
            </div>
            <button className="place-card__bookmark-button button" type="button">
              <svg className="place-card__bookmark-icon" width="18" height="19">
                <use xlinkHref="#icon-bookmark"></use>
              </svg>
              <span className="visually-hidden">To bookmarks</span>
            </button>
          </div>
          <div className="place-card__rating rating">
            <div className="place-card__stars rating__stars">
              <span style={{width: getRating(apartment.rating)}}></span>
              <span className="visually-hidden">Rating</span>
            </div>
          </div>
          <h2 className="place-card__name">
            <a href="#" onClick={onClick}>{ apartment.title }</a>
          </h2>
          <p className="place-card__type">{apartment.type}</p>
        </div>
      </article>
    );
  }
}

Apartment.propTypes = {
  apartment: apartmentProps,
  onClick: PropTypes.func.isRequired,
  onMouseOver: PropTypes.func.isRequired,
  onMouseOut: PropTypes.func.isRequired
};

export default Apartment;

