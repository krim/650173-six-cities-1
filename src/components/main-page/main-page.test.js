import React from 'react';
import renderer from 'react-test-renderer';

import apartment from '../../__fixtures__/apartment';
import mapBuilder from '../../mocks/map-builder';
import MainPage from './main-page.jsx';

const apartments = [apartment];
const city = apartment.city;

describe(`MainPage`, () => {
  it(`renders component correctly`, () => {
    const tree = renderer.create(
        <MainPage
          apartments={apartments}
          city={city}
          cities={[city]}
          mapSettings={
            {builder: mapBuilder, zoomControl: false, marker: true}
          }
          switchCity={jest.fn()}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
