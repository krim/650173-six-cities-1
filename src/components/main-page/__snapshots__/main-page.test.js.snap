// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage without apartments renders component correctly 1`] = `
<main
  className="page__main page__main--index"
>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <div
    className="cities tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      >
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item tabs__item--active"
            href="#"
            onClick={[Function]}
          >
            <span>
              Amsterdam
            </span>
          </a>
        </li>
      </ul>
    </section>
  </div>
  <div
    className="cities__places-wrapper"
  >
    <div
      className="cities__places-container container"
    >
      <section
        className="cities__places places"
      >
        <h2
          className="visually-hidden"
        >
          Places
        </h2>
        <b
          className="places__found"
        >
          1
           places to stay in 
          Amsterdam
        </b>
        <form
          action="#"
          className="places__sorting"
          method="get"
        >
          <span
            className="places__sorting-caption"
          >
            Sort by 
          </span>
          <span
            className="places__sorting-type"
            onClick={[Function]}
            tabIndex="0"
          >
            Popular
            <svg
              className="places__sorting-arrow"
              height="4"
              width="7"
            >
              <use
                xlinkHref="#icon-arrow-select"
              />
            </svg>
          </span>
          <ul
            className="places__options places__options--custom"
          >
            <li
              className="places__option places__option--active"
              onClick={[Function]}
              tabIndex="0"
            >
              Popular
            </li>
            <li
              className="places__option "
              onClick={[Function]}
              tabIndex="1"
            >
              Price: low to high
            </li>
            <li
              className="places__option "
              onClick={[Function]}
              tabIndex="2"
            >
              Price: high to low
            </li>
            <li
              className="places__option "
              onClick={[Function]}
              tabIndex="3"
            >
              Top rated first
            </li>
          </ul>
        </form>
        <div
          className="cities__places-list places__list tabs__content"
        >
          <article
            className="cities__place-card place-card"
          >
            <div
              className="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="https://es31-server.appspot.com/six-cities/static/hotel/1.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    393
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "60%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  Waterfront with extraordinary view
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                Hotel
              </p>
            </div>
          </article>
        </div>
      </section>
      <div
        className="cities__right-section"
      >
        <section
          className="cities__map map"
          id="map"
        />
      </div>
    </div>
  </div>
</main>
`;

exports[`MainPage without apartments renders component correctly 2`] = `
<main
  className="page__main page__main--index page__main--index-empty"
>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <div
    className="cities tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      >
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item tabs__item--active"
            href="#"
            onClick={[Function]}
          >
            <span>
              Amsterdam
            </span>
          </a>
        </li>
      </ul>
    </section>
  </div>
  <div
    className="cities__places-wrapper"
  >
    <div
      className="cities__places-container cities__places-container--empty container"
    >
      <section
        className="cities__no-places"
      >
        <div
          className="cities__status-wrapper tabs__content"
        >
          <b
            className="cities__status"
          >
            No places to stay available
          </b>
          <p
            className="cities__status-description"
          >
            We could not find any property availbale at the moment in Dusseldorf
          </p>
        </div>
      </section>
      <div
        className="cities__right-section"
      />
    </div>
  </div>
</main>
`;
